### PR TITLE
p2p: delete legacy stack initial pass

### PR DESCRIPTION
**Pull request by tychoish**
A few notes:

- this is not all the deletion that we can do, but this is the most
  "simple" case: it leaves in shims, and there's some trivial
  additional cleanup to the transport that can happen but that
  requires writing more code, and I wanted this to be easy to review
  above all else.
  
- This should land *after* we cut the branch for 0.35, but I'm
  anticipating that to happen soon, and I wanted to run this through
  CI.

### CODE REVIEW

File: internal/p2p/shim.go [hunk 1]
```diff
@@ -1,58 +1,42 @@
...
-	// responsible for wiring up legacy p2p behavior to the new p2p semantics
-	// (e.g. proxying Envelope messages to legacy peers).
-	ReactorShim struct {
-		BaseReactor
-
-		Name        string
-		PeerUpdates *PeerUpdates
-		Channels    map[ChannelID]*ChannelShim
-	}
+// ChannelDescriptorShim defines a shim wrapper around a legacy p2p channel
```
Comment (creachadair) [inline internal/p2p/shim.go:12]: Should we use "Legacy" in the name? Also possibly document that this is intended to go away, though if this is just the first step of more to come, I'm not deeply worried about either of those.

File: internal/p2p/shim.go [hunk 2]
```diff
@@ -93,121 +74,15 @@ func NewChannelShim(cds *ChannelDescriptorShim, buf uint) *ChannelShim {
...
-// GetChannel returns a p2p Channel reference for a given ChannelID. If no
-// Channel exists, nil is returned.
-func (rs *ReactorShim) GetChannel(cID ChannelID) *Channel {
-	channelShim, ok := rs.Channels[cID]
-	if ok {
-		return channelShim.Channel
-	}
-
-	return nil
+// MConnConfig returns an MConnConfig with fields updated
```
Comment (creachadair) [inline internal/p2p/shim.go:77]:
```suggestion
// MConnConfig returns an MConnConfig based on the defaults, with fields updated
```

File: internal/p2p/pex/reactor.go [hunk 3]
```diff
@@ -24,6 +24,22 @@ var (
...
+	// over-estimate of max NetAddress size
+	// hexID (40) + IP (16) + Port (2) + Name (100) ...
+	// NOTE: dont use massive DNS name ..
+	maxAddressSize = 256
+
+	// max addresses returned by GetSelection
+	// NOTE: this must match "maxMsgSize"
+	maxGetSelection = 250
+
+	// NOTE: amplificaiton factor!
```
Comment (creachadair) [inline internal/p2p/pex/reactor.go:39]:
```suggestion
	// NOTE: amplification factor!
```

File: internal/p2p/router.go [hunk 4]
```diff
@@ -21,6 +21,8 @@ import (
 
 const queueBufferDefault = 32
 
+const dialRandomizerIntervalMilliseconds = 3000
```
Comment (creachadair) [inline internal/p2p/router.go:24]: Maybe:
```suggestion
const dialRandomizerInterval = 3000*time.Millisecond
```

?

Though I'm not seeing where this is used at the moment.

File: internal/rpc/core/consensus.go [hunk 5]
```diff
@@ -58,52 +54,28 @@ func (env *Environment) DumpConsensusState(ctx *rpctypes.Context) (*coretypes.Re
...
+			continue
+		}
+
+		peerStateJSON, err := peerState.ToJSON()
+		if err != nil {
+			return nil, err
+		}
+
+		addr := env.PeerManager.Addresses(pid)
+		if len(addr) >= 1 {
```
Comment (creachadair) [inline internal/rpc/core/consensus.go:71]: Optional nit.
```suggestion
		if len(addr) != 0 {
```

File: test/e2e/generator/generate_test.go [hunk 6]
```diff
@@ -5,15 +5,14 @@ import (
...
 	"github.com/stretchr/testify/require"
 	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
 )
 
 func TestGenerator(t *testing.T) {
-	manifests, err := Generate(rand.New(rand.NewSource(randomSeed)), Options{P2P: MixedP2PMode})
+	manifests, err := Generate(rand.New(rand.NewSource(randomSeed)), Options{})
 	require.NoError(t, err)
-	require.True(t, len(manifests) >= 64, "insufficient combinations")
+	require.True(t, len(manifests) >= 24, "insufficient combinations %d", len(manifests))
```
Comment (creachadair) [inline test/e2e/generator/generate_test.go:15]: Yay!

**Review by creachadair — APPROVED** [review]
This seems like a nice clean cut, I like it a lot. None of my comments are that important, just some suggestions.

File: internal/p2p/pex/reactor.go [hunk 7]
```diff
@@ -24,6 +24,22 @@ var (
 // TODO: Consolidate with params file.
 // See https://github.com/tendermint/tendermint/issues/6371
 const (
+	// PexChannel is a channel for PEX messages
+	PexChannel = byte(0x00)
```
[the remark this answers is not in the document]
Comment (tychoish) [reply internal/p2p/pex/reactor.go:28]: I think I want to revisit channel registration and the way that impacts node initialization as part of something more considered, and probably make this private anyway

File: internal/p2p/shim.go (hunk 1 above)
Comment (tychoish) [reply internal/p2p/shim.go:12]: yeah, this is just transient, the shims and channel registration will get deleted in the second pass, but it needs a little more fineness than I wanted to do in this PR. just to be clear: this file goes away, and channel registration will get ver

**Comment by faddat** [comment]
thanks.

**Review by marbar3778 — APPROVED** [review]
🥳

**Review by cmwaters — CHANGES_REQUESTED** [review]
Looks good. I'm feeling better already at the clean up. There will need to be a few follow up changes but I think it's fine to break this up in the way you have.

I would request that you remove the deletion of the v2 blocksync reactor from this PR. It should be done as its own PR with comments explaining the rationale for future users why this happened. I would ideally like to do some form of post mortem where we analyse what prompted there to be a v2 reactor in the first place, what was it trying to solve, what went wrong and is there anything we can take from it to improve the v0 reactor.

**Comment by tychoish** [comment]
This must merge *after* #7046, to make sure that blocksync/v2 gets removed in a separate commit (because blocksync/v2 can't exist in the code base without the legacy p2p layer.)

**Review by creachadair — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]
LGTM 👍  

I think we should also add a changelog entry

**Pull request merged**